contents/java/Spring.md

```diff
@@ -0,0 +1,193 @@
+This chapter assumes that the reader has a basic knowledge of Java enterprise.
+</box>
+
+<box id="article-toc">
+
+* [What is Spring Framework‎](#what-is-spring-framework)
+* [Core Features](#core-features)
+  * [The IOC Container](#the-ioc-container)
+  * [Spring JDBCTemplate](#spring-jdbctemplate)
+  * [Spring AOP](#spring-aop)
+* [Why use Spring‎](#why-use-spring)
+* [How to get started‎](#how-to-get-started)
+</box>
+
+## What is Spring Framework?
+
+The [official website](https://docs.spring.io/spring/docs/current/spring-framework-reference/overview.html) describes Spring Framework as follows:
+
+>*Spring Framework* is an open-source framework to create Java enterprise applications.
+>It provides foundational support for different application architectures, including messaging, transactional data and persistence, and web.
+
+## Core Features
+
+### The IoC Container
+
+The Spring IoC container is at the core of the Spring Framework. The container will create the objects, wire them together,
+configure them, and manage their complete life cycle from creation till destruction. The Spring container uses dependency injection (DI)
+to manage the components that make up an application.
+    
+In the Spring framework, the IoC container is represented by the interface ApplicationContext. In order to assemble beans,
+the container uses configuration metadata, which can be in the form of XML configuration or annotations.
+
+Below is one way to manually instantiate a container:
+```
+    ApplicationContext context = new ClassPathXmlApplicationContext("applicationContext.xml");
+```
+   
+We can then perform dependency injection by using the external XML file.
+It makes our code loosely coupled and easier for testing. For example, we can have a simple Employee class
+containing two fields id and name with its setters and getters and a method to display these informations.
+```java
+    package com.company;  
+  
+    public class Employee {  
+       private int id;  
+       private String name;  
+       private String city;  
+  
+       public int getId() {  
+           return id;  
+       }  
+       public void setId(int id) {  
+           this.id = id;  
+       }  
+       public String getName() {  
+           return name;  
+       }  
+       public void setName(String name) {  
+           this.name = name;  
+       }  
+       void display(){  
+           System.out.println(id+" "+name);  
+       }
+    }
+```
+    
+And the corresponding `applicationContext.xml` file could be:
+```
+    <?xml version="1.0" encoding="UTF-8"?>  
+    <beans  
+    xmlns="http://www.springframework.org/schema/beans"  
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"  
+    xmlns:p="http://www.springframework.org/schema/p"  
+    xsi:schemaLocation="http://www.springframework.org/schema/beans  
+                http://www.springframework.org/schema/beans/spring-beans-3.0.xsd">  
+  
+    <bean id="obj" class="com.company.Employee">  
```

Comment:
Would it be better to add some indentation here so that it is easier for the reader to understand? Also maybe can omit the xml namespaces to only highlight the important part?